iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/MetastoreUtil.java

```diff
@@ -87,4 +103,39 @@ public static void alterTable(
       }
     }
   }
+
+  public static Table convertIcebergTableToHiveTable(org.apache.iceberg.Table icebergTable, Configuration conf) {
+    Table hiveTable = new Table();
+    TableMetadata metadata = ((BaseTable) icebergTable).operations().current();
+    long maxHiveTablePropertySize = conf.getLong(HiveOperationsBase.HIVE_TABLE_PROPERTY_MAX_SIZE,
+        HiveOperationsBase.HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+    HMSTablePropertyHelper.updateHmsTableForIcebergTable(metadata.metadataFileLocation(), hiveTable, metadata,
+        null, true, maxHiveTablePropertySize, null);
+    hiveTable.getParameters().put(CatalogUtils.ICEBERG_CATALOG_TYPE, CatalogUtil.ICEBERG_CATALOG_TYPE_REST);
```

Comment:
hardcode - ICEBERG_CATALOG_TYPE_REST

Reply:
is `iceberg.catalog.default_iceberg.type` config used by the Iceberg RestCatalog? have you tried Polaris or Gravitino?

Reply:
Iceberg RESTCatalog uses catalog type as input ("type" property) when initializing a catalog, but when it returns a table its properties don't include any catalog properties. Looks like RESTCatalog does not use this property: `iceberg.catalog.default_iceberg.type`